src/main/java/com/assignment/application/controller/ProjectController.java

```diff
@@ -13,22 +16,68 @@
-                                                @PathVariable("comp_id") Long companyId){
-        return projectServiceI.deleteProject(projectId,companyId);
+                                                @PathVariable("comp_id") Long companyId,
+                                                @RequestHeader("username") String username,
+                                                @RequestHeader("password") String password) {
+        try {
+            int status = verifyUser.authorizeUser(username, password);
+            if (status == 0) {
+                return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
+            }
+            if (projectServiceI.deleteProject(projectId, companyId).equalsIgnoreCase(StringConstant.DELETION_SUCCESSFUL)) {
+                return new ResponseEntity<>(StringConstant.DELETION_SUCCESSFUL, HttpStatus.OK);
+            }
+            return new ResponseEntity<>(StringConstant.INVALID_CREDENTIALS, HttpStatus.BAD_REQUEST);
+        } catch (Exception e) {
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+        }
     }
 
-    @PostMapping(value="")
-    public ResponseEntity<Project> addProject(@PathVariable("comp_id") Long companyId,@RequestBody Project project){
-        return projectServiceI.addCompProject(companyId,project);
+    @PostMapping(value = "")
+    public ResponseEntity<Project> addProject(@PathVariable("comp_id") Long companyId,
+                                              @RequestBody Project project,
+                                              @RequestHeader("username") String username,
+                                              @RequestHeader("password") String password) {
+        try {
+            int status = verifyUser.authorizeUser(username, password);
+            if (status == 0) {
+                return new ResponseEntity<>(null, HttpStatus.UNAUTHORIZED);
+            }
+            Project projectToBeAdded = projectServiceI.addCompProject(companyId, project);
+            if (projectToBeAdded == null) {
+                return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+            }
+            return new ResponseEntity<>(projectToBeAdded, HttpStatus.OK);
+        } catch (Exception e) {
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
+        }
+
     }
 
-    @GetMapping(value="")
-    public ResponseEntity<List<Project>> getProject(@PathVariable("comp_id") Long companyId){
-        return projectServiceI.getProject(companyId);
+    @GetMapping(value = "")
+    public ResponseEntity<List<Project>> getProject(@PathVariable("comp_id") Long companyId,
+                                                    @RequestHeader("username") String username,
+                                                    @RequestHeader("password") String password) {
+        try {
+            int status = verifyUser.authorizeUser(username, password);
```

Comment:
Or instead we could have written a controller advice with exception handling for auth errors